### ISSUE TITLE
Compile OCLint on FreeBSD 10.3

### ISSUE THREAD

**Issue by capjo** [oclint/oclint#396]
The make scripts that come with oclint fail during the build process on FreeBSD 10.3. LLVM 3.9.1 gets build but later on the script fails with:

Scanning dependencies of target OCLintRuleSet
[  7%] Building CXX object lib/CMakeFiles/OCLintRuleSet.dir/RuleConfiguration.cpp.o
[ 15%] Building CXX object lib/CMakeFiles/OCLintRuleSet.dir/RuleSet.cpp.o
In file included from /usr/home/ah/tools/oclint/oclint-core/lib/RuleConfiguration.cpp:1:
/usr/home/ah/tools/oclint/oclint-core/include/oclint/RuleConfiguration.h:4:10: fatal error: 'string' file not found
#include <string>

**Comment by capjo** [comment]
The issue is in: oclint/oclint/blob/master/oclint-core/cmake/OCLintConfig.cmake

It assumes libstdc++ as default for non APPLE 

IF(APPLE)
    SET(CMAKE_CXX_FLAGS "-std=c++11 -stdlib=libc++ ${CMAKE_CXX_FLAGS}")
ELSEIF(MINGW)
    SET(CMAKE_CXX_FLAGS "-std=gnu++11 ${CMAKE_CXX_FLAGS}")
ELSE()
    IF(CMAKE_CXX_COMPILER_ID MATCHES "**Clang**")
        SET(CMAKE_CXX_FLAGS "**-stdlib=libstdc++** ${CMAKE_CXX_FLAGS}")
    ENDIF()
    SET(CMAKE_CXX_FLAGS "-std=c++11 ${CMAKE_CXX_FLAGS}")
ENDIF()

**Comment by ryuichis** [comment]
@capjo right, we supported freebsd in earlier versions, but due to the low demand at that time, we cleaned up some specific configurations for freebsd environments. Are you interested in providing a pull request to make it compile again on freebsd? (You might be able to look into that cmake file's commit history to find some outdated configurations.) And since we do have analytics module in place at this time, we can for sure understand how OCLint is used in freebsd. 

Another related question, are you aware of any cloud-based CI tools, like Travis CI, that run on top of freebsd os?
